var/spack/repos/builtin/packages/pcre2/package.py

```diff
@@ -30,7 +30,30 @@ class Pcre2(AutotoolsPackage):
 
     variant("multibyte", default=True, description="Enable support for 16 and 32 bit characters.")
     variant("jit", default=False, description="enable Just-In-Time compiling support")
+    variant("shared", default=True, description="build shared pcre2", when="build_system=cmake")
```

Comment:
It seems a bit strange to constrain this variant to the cmake build system.

* I see `pcre2` has `--enable-shared` for Autotools
* or perhaps this doesn't need to be a variant on Windows? e.g. if all dependents can make do with one or another (at least, this seems to have been the idea for Linux/Mac)

Reply:
The reason I added this explicitly for CMake is because by default the CMake system builds both static and shared libraries, which, in addition to being bad practice, is a recipe for disaster on Windows and makes lib detection a bit nebulous, particularly when you have to reason about whether to append `-static` to the lib names. I believe the autotools build _also_ builds static and shared libraries, so you always get both, which is perfectly fine/normal on systems that support autotools, but means that we don't necessarily need to provide this variant.

Maybe we do want to make it Windows only as the dual binaries are perfectly acceptable on other platforms.

Reply:
> Although I assume this could also be resolved by updating .libs in pcre2

`.libs` was updated by this PR, to avoid just providing static (or shared) only. Depending on the dependent package build type this may need to be either static or shared, so always returning one means that's the only build type allowed when consuming pcre2 on Windows. I don't know how we would update 

> But assuming we want to force building one or the other, we can also arbitrarily choose one if we like

Arbitrarily guessing what the calling project wants/needs seems like a fragile path forward as opposed to allowing dependent packages to express what they need. For example we got into this change because the Qt build expected a static pcre2, got a shared, and couldn't link against the mangled symbol names because it was looking for the unmangled names.
That also seems to go against what I understand of the Spack idiom where we avoid making choices for the user/dependents in a completely obfuscated manner.

Reply:
I'm happy to make that change (and I will now regardless of your response to this comment) but I was actually starting to lean the other way, away from what I originally said, to say that we should make this a variant everywhere. IMO if we have the option to make a package more flexible, we should do that, and this variant gives users/dependents the ability to specify a requirement for shared or static libs, at no real cost, it's not like we can use both binaries at once anyway, and do to the configuration of the libs method, nothing can actually take advantage of the static build anyway without a non trivial amount of manual lift.